internal/mux/mux.go

```diff
@@ -41,6 +42,9 @@ func New(channelNum, bitDepthInBytes int) *Mux {
 	return m
 }
 
+// Read reads data from all of its readers, interprets it as samples with the bit depth
+// specified during its creation, then adds all of the samples together and fills the buf
+// slice with the result of this.
 func (m *Mux) Read(buf []byte) (int, error) {
```

Comment:
Option: I'd want to note that if there are no readers, Mux fulfills some zeros to buf in order to avoid freezing by e.g. io.ReadAll.

Reply:
Okay, I'll add it.

Reply:
Fixed!

Reply:
> in case all readers got drained before fully filling buf,

I don't think I implemented such thing?

> Read fills the rest of buf with zeros.

Read doesn't fill all the rest of buf with zeros. In the current implementation, this fills just 256 bytes at most :)

Reply:
I corrected that it never drains. It drains when it gets closed.

Reply:
Aha, okay, I'll fix that.

Reply:
Done!